packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,513 @@
+      padding: _kHorizontalItemPadding.resolve(Directionality.of(context)),
+      child: UnconstrainedBox(
+        constrainedAxis: Axis.horizontal,
+        child: box,
+      ),
+    );
+  }
+}
+
+class _SegmentedControlRenderWidget<T> extends MultiChildRenderObjectWidget {
+  _SegmentedControlRenderWidget({
+    Key key,
+    List<Widget> children = const <Widget>[],
+    @required this.selectedIndex,
+    @required this.pressedIndex,
+  }) : super(
+          key: key,
+          children: children,
+        );
+
+  final int selectedIndex;
+  final int pressedIndex;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderSegmentedControl<T>(
+      textDirection: Directionality.of(context),
+      selectedIndex: selectedIndex,
+      pressedIndex: pressedIndex,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderSegmentedControl<T> renderObject) {
+    renderObject
+      ..textDirection = Directionality.of(context)
+      ..selectedIndex = selectedIndex
+      ..pressedIndex = pressedIndex;
+  }
+}
+
+class _SegmentedControlContainerBoxParentData extends ContainerBoxParentData<RenderBox> {
+  RRect surroundingRect;
+}
+
+class _RenderSegmentedControl<T> extends RenderBox with
+    ContainerRenderObjectMixin<RenderBox, ContainerBoxParentData<RenderBox>>,
+    RenderBoxContainerDefaultsMixin<RenderBox, ContainerBoxParentData<RenderBox>>{
+  _RenderSegmentedControl({
+    List<RenderBox> children,
+    @required int selectedIndex,
+    @required int pressedIndex,
+    @required TextDirection textDirection,
+  })  : _textDirection = textDirection,
+        _selectedIndex = selectedIndex,
+        _pressedIndex = pressedIndex {
+    addAll(children);
+  }
+
+  int get selectedIndex => _selectedIndex;
+  int _selectedIndex;
+  set selectedIndex(int value) {
+    if (_selectedIndex == value) {
+      return;
+    }
+    _selectedIndex = value;
+    markNeedsPaint();
+  }
+
+  int get pressedIndex => _pressedIndex;
+  int _pressedIndex;
+  set pressedIndex(int value) {
+    if (_pressedIndex == value) {
+      return;
+    }
+    _pressedIndex = value;
+    markNeedsPaint();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value) {
+      return;
+    }
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  List<Paint> get fillPaint => _fillPaint;
+  final List<Paint> _fillPaint = <Paint>[];
+
+  @override
+  double computeMinIntrinsicWidth(double height) => constraints.maxWidth / childCount;
+
+  @override
+  double computeMaxIntrinsicWidth(double height) => constraints.maxWidth / childCount;
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    RenderBox child = firstChild;
+    double minHeight = double.infinity;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
```

Comment:
You could precompute `constraints.maxWidth / childCount` here to avoid an extra divide.